src/e2e/java/teammates/e2e/cases/AdminNotificationsPageE2ETest.java

```diff
@@ -0,0 +1,89 @@
+package teammates.e2e.cases;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.common.util.TimeHelper;
+import teammates.e2e.pageobjects.AdminNotificationsPage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#ADMIN_NOTIFICATIONS_PAGE}.
+ */
+public class AdminNotificationsPageE2ETest extends BaseE2ETestCase {
+    private NotificationAttributes[] notifications = new NotificationAttributes[2];
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/AdminNotificationsPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        notifications[0] = testData.notifications.get("notification1");
+        notifications[1] = testData.notifications.get("notification2");
+    }
+
+    @Test
+    @Override
+    public void testAll() {
+
+        ______TS("verify loaded data");
+        AppUrl url = createFrontendUrl(Const.WebPageURIs.ADMIN_NOTIFICATIONS_PAGE);
+        AdminNotificationsPage notificationsPage = loginAdminToPage(url, AdminNotificationsPage.class);
+        // Only validates that the notifications are present in the notifications table instead of checking every row
+        // This is because the page will display all notifications in the database, which is not predictable
+        notificationsPage.verifyNotificationsTableRow(notifications[0]);
+        notificationsPage.verifyNotificationsTableRow(notifications[1]);
+        verifyPresentInDatabase(notifications[0]);
+        verifyPresentInDatabase(notifications[1]);
+
+        ______TS("add new notification");
+        NotificationAttributes newNotification = NotificationAttributes
+                .builder("placeholder-notif-id")
+                .withStartTime(TimeHelper.parseInstant("2035-04-01T22:00:00Z"))
+                .withEndTime(TimeHelper.parseInstant("2035-04-30T20:00:00Z"))
+                .withStyle(NotificationStyle.SUCCESS)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withTitle("E2E test notification 1")
+                .withMessage("<p>E2E test notification message</p>")
+                .build();
+
+        notificationsPage.addNotification(newNotification);
+        notificationsPage.verifyStatusMessage("Notification created successfully.");
+
+        // Replace placeholder ID with actual ID of created notification
+        notificationsPage.sortNotificationsTableByDescendingCreateTime();
+        String newestNotificationId = notificationsPage.getFirstRowNotificationId();
+        newNotification.setNotificationId(newestNotificationId);
```

Comment:
> // Replace placeholder ID with actual ID of created notification

Might not be necessary as `verifyPresentInDatabase` will do that for you when it calls `verifyEquals`. In fact, besides the id, the create and update time does not matter as well.

```
private void equalizeIrrelevantData(NotificationAttributes expected, NotificationAttributes actual) {
        // Ignore time field as it is stamped at the time of creation in testing
        expected.setCreatedAt(actual.getCreatedAt());
        expected.setUpdatedAt(actual.getUpdatedAt());

        // Ignore ID field as it is unique - UUID
        expected.setNotificationId(actual.getNotificationId());
    }
```

Reply:
After doing some testing, I think replacing the ID is still necessary as we need the ID to get the notification from the database before we call `verifyEquals`. Will be making a commit to remove the comment though!

Reply:
This feels like a chicken and egg problem, because in order for `notificationsPage.getFirstRowNotificationId()` to return the newest id, we need to be sure that the newly added notification has already persisted into DB and reflected in the frontend. But at that point we cannot be sure because `verifyPresentInDatabase()` is not called yet and to call `verifyPresentInDatabase()` we need the new notification's id.

Reply:
From `admin-notifications-page.component.ts`, in `addNewNotificationHandler()`:
```
this.notificationService.createNotification(
    ...
    .subscribe(
      (notification: Notification) => {
        this.statusMessageService.showSuccessToast('Notification created successfully.');

        this.notificationsTableRowModels.unshift({
          isHighlighted: true,
          notification,
        });
     ...
```

Currently, after clicking the create notification button, the frontend makes the POST request, shows the success toast, and adds the notification returned to the notifications table. If I change the order such that the success toast is shown after the notification is added to the table, would checking for the success toast be sufficient in making sure that the newly added notification is reflected in the frontend?

Reply:
Good point! `notificationsPage.verifyStatusMessage()` would've checked that the notification is added successfully, so my initial concern is invalid. Not sure whether we can rely on that, but I guess it's better than nothing, and so far the E2E tests are passing which is a good sign that probably it works? If you really want you can switch up the order but I doubt it makes a difference.

Reply:
> I believe this also means `expected.setNotificationId(actual.getNotificationId());` in `equalizeIrrelevantData` is redundant, so should I remove this line altogether?

If it does not affect the component tests which also uses `verifyPresentInDatabase` we can remove.